test/benchmark.c

```diff
@@ -3,17 +3,110 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
-#include <unistd.h>
+#ifndef _WIN32
+	#include <unistd.h>
+#else
+	#include <io.h>
+#endif
 #include <stdlib.h>
 #include <stdio.h>
 #include <time.h>
+#if defined(_WIN32)
+	#include <windows.h>
+	#include <BaseTsd.h>
+	#include <Wincrypt.h>
 
+	typedef SSIZE_T ssize_t;
+#endif
 #include "../include/libbase64.h"
 #include "codec_supported.h"
 
 #define INSIZE_MB  10
 #define RANDOMDEV  "/dev/urandom"
 
+#ifdef _WIN32
+typedef unsigned int clockid_t;
+
+#ifndef CLOCK_MONOTONIC
+#define CLOCK_MONOTONIC 1
+#endif
+
+#ifndef CLOCK_REALTIME
+#define CLOCK_REALTIME 2
+#endif
+
+/* Number of 100 ns intervals from January 1, 1601 till January 1, 1970. */
+const static unsigned long long unix_epoch = 116444736000000000;
+struct timespec {
+	time_t tv_sec;
+	long tv_nsec;
+};
+static ULARGE_INTEGER
+xgetfiletime(void) {
+	ULARGE_INTEGER current_time;
+	FILETIME current_time_ft;
+
+	/* Returns current time in UTC as a 64-bit value representing the number
+    * of 100-nanosecond intervals since January 1, 1601 . */
+	GetSystemTimePreciseAsFileTime(&current_time_ft);
+	current_time.LowPart = current_time_ft.dwLowDateTime;
+	current_time.HighPart = current_time_ft.dwHighDateTime;
+
+	return current_time;
+}
+
+static int 
+clock_gettime(clock_t id, struct timespec* ts) {
+	if (id == CLOCK_MONOTONIC) {
+		static LARGE_INTEGER freq;
+		LARGE_INTEGER count;
+		long long int ns;
+
+		if (!freq.QuadPart) {
+			/* Number of counts per second. */
+			QueryPerformanceFrequency(&freq);
+		}
+		/* Total number of counts from a starting point. */
+		QueryPerformanceCounter(&count);
+
+		/* Total nano seconds from a starting point. */
+		ns = (double)count.QuadPart / freq.QuadPart * 1000000000;
+
+		ts->tv_sec = count.QuadPart / freq.QuadPart;
+		ts->tv_nsec = ns % 1000000000;
+	}
+	else if (id == CLOCK_REALTIME) {
+		ULARGE_INTEGER current_time = xgetfiletime();
+
+		/* Time from Epoch to now. */
+		ts->tv_sec = (current_time.QuadPart - unix_epoch) / 10000000;
+		ts->tv_nsec = ((current_time.QuadPart - unix_epoch) %
+			10000000) * 100;
+	}
+	else {
+		return -1;
+	}
+
+	return 0;
+}
+
+int 
+crypt_rand(void* data, size_t length) {
+	HCRYPTPROV hProvider = 0;
+
+	if (!CryptAcquireContext(&hProvider, 0, 0, PROV_RSA_FULL, CRYPT_VERIFYCONTEXT | CRYPT_SILENT))
+		return 1;
+	if (!CryptGenRandom(hProvider, length, data)) {
+		CryptReleaseContext(hProvider, 0);
+		return 1;
+	}
+	if (!CryptReleaseContext(hProvider, 0))
+		return 1;
+	return 0;
+}
+
+#endif /* _WIN32 */
+
```

Comment:
I would consider splitting the implementation of `get_random_data()` into two separate files, `rand_win.c` and `rand_unix.c` and `#include`'ing one of them at compile time based on whether `_WIN32` is defined. I am not a fan of elaborate preprocessor trees.